### ISSUE TITLE
New highlight.js website! Help us test it :smile:

### ISSUE THREAD

**Issue by allejo** [highlightjs/highlight.js#3611]
Hello everyone! I'm by no means someone you see hanging out in this repo often; I stop by whenever code reviews are needed or a second opinion is wanted. Something I've been working on for months now (on and off) is a new website for our beloved syntax highlighting library 🥳

Our new website is powered by [Next.js](https://nextjs.org/) and is generously hosted by [Heroku](https://www.heroku.com/). We need your help testing the new website, giving us feedback, reporting bugs, etc.

I've tried my best to meet feature parity with the current website, reorganizing some pages, and adding a much-requested demo page!

Website: https://highlightjs-web.herokuapp.com/
GitHub repository: https://github.com/highlightjs/highlightjs.org
Report issues here: https://github.com/highlightjs/highlightjs.org/issues

Ideally, we'd like to go live with the new website towards the end of September, early October.

## Administrative Tasklist

- [ ] Switch DNS for `https://highlightjs.org/` to point to Heroku
- [ ] Build Github Workflows for new highlight.js releases to trigger a new website build

/cc @highlightjs/core

**Comment by taufik-nurrohman** [comment]
FYI, the language pack generator feature is very useful. Wish there will be a public API that I could use to generate my own language set through my application. An API URL that simply return a plain text response that I could save as `highlight.js` for example.

**Comment by allejo** [comment]
> FYI, the language pack generator feature is very useful. Wish there will be a public API that I could use to generate my own language set through my application. An API URL that simply return a plain text response that I could save as `highlight.js` for example.

Underneath the new `/download` page lives an undocumented (as of this moment) `/api/download` endpoint that accepts POST requests (watch your browser's Network tab) with a JSON body in the structure of,


```json
{
  "api": 2,
  "languages": []
}
```

This API endpoint returns an `attachment` response with the bundled zip file. I'm open to considering adding an endpoint that returns just the JS and not the entire zip.

**Comment by joshgoebel** [comment]
I'd suggest perhaps a `format: [zip,javascript]` choice perhaps?

**Comment by joshgoebel** [comment]
Some indication that "Docs" and "GitHub" were going to jump you to another site might be nice... I always frown when I see links in "sitewide navigation" that link to entirely different sites without any warning.  Maybe move them to right with a `|` inbetween with a diff color/shade?

**Comment by isagalaev** [comment]
This looks really nice :-) Glad you kept the color scheme!

**Comment by joshgoebel** [comment]
> This looks really nice 

Yes, it looks amazing.  I totally should have led with that before jumping into my [overall] minor nitpicks... I'd be fine launching it just as soon as we hook up the Github actions to deal with new releases... and of course we can continue to improve post-launch as well... or I'm also ok with the plan to test a bit longer (or wait for more feedback), then launch.  

@allejo Were you planning to help setup the GH actions also?  I think some of the work we've already done with auto-publishing the cdn-release gem might be relevant?  I imagine we're talking two parts:

- GH action here pushes to Heroku on now commits, etc
- GH release on main repo updates this repo's "source" files? (or bumps a version or something)

I haven't looked too deeply into the programming here yet.
-

**Comment by allejo** [comment]
> Does VS Code use us directly or just in plugins? If we're not in the default install I think that's a little deceptive and we should remove them from the list and find another big site.

That's true 🤔 As far as I know, VSC uses hljs for syntax highlighting of markdown code blocks in previews. I'd welcome any other websites/services that use us to replace VSC!

> Were you planning to help setup the GH actions also? I think some of the work we've already done with auto-publishing the cdn-release gem might be relevant? 

Yes! I'll be looking into the GitHub actions for syncing releases. I think there will definitely be shared logic between these actions

> * GH action here pushes to Heroku on now commits, etc

Heroku provides a GitHub app that syncs stuff for us, this is already taken care of 🥳 

> * GH release on main repo updates this repo's "source" files? (or bumps a version or something)

Correct, it'd bump the `package.json` file in the website's repo to the latest release and rebuild everything from there.

**Comment by joshgoebel** [comment]
https://github.com/highlightjs/highlight.js/blob/main/.github/workflows/release.yml#L82

This is where the relevant part starts...

**Comment by Hirse** [comment]
As someone who's in the past contributed to the (old? current?) website, the new one is **looking nice**! 

A few things I noticed:
- different favicon, is that intentionally?
- Links to Docs and GitHub not opening in new tab
- Add link to npm?
- Update default theme (or add new default-dark) to use on the website?
- "News" -> "Releases"?
- I like the design of the news, but it doesn't really match the rest of the page
- Why is "Code Samples" such a tiny link in the bottom?

Happy to contribute for the above. 😄

**Comment by adsingh14** [comment]
Good to see this! I've noticed one bug,

1. Visit https://highlightjs-web.herokuapp.com/download

2. Type 'test' 

-  It'll show blank box

3. Again type 'html'. That blank box will stay there (stick with the header)

**Comment by AdamRaichu** [comment]
As of right now, https://highlightjs-web.herokuapp.com is not working.

**Comment by allejo** [comment]
> As of right now, https://highlightjs-web.herokuapp.com is not working.

Life really got in the way with my commitments, especially to this project... I will be working on the logistics of migrating the site over the next week or so 🤞

The website demo can now be found at: https://hljs.netlify.app

I would really appreciate any help testing and finding bugs!

**Comment by jf990** [comment]
@allejo I did some ad-hoc testing and everything looks great to me, really nice work. The only issue I ran into was the [download](https://hljs.netlify.app/download), even though I selected a small subset of languages, my download contained what looked like all languages.

**Comment by allejo** [comment]
> The only issue I ran into was the [download](https://hljs.netlify.app/download), even though I selected a small subset of languages, my download contained what looked like all languages.

Fixed in https://github.com/highlightjs/highlightjs.org/commit/ebf0ac54d5f41f9078e4e0c99dac0be7568b3a87

**Comment by allejo** [comment]
> * different favicon, is that intentionally?

not intentional! I always forget about favicons 🤦‍♂️ fixed in https://github.com/highlightjs/highlightjs.org/commit/9e225ea9de5d6086183238038f1c0aee4b7680a4

> * Links to Docs and GitHub not opening in new tab

fixed in https://github.com/highlightjs/highlightjs.org/commit/93e1acb31c4427927a05645c9a573858ebce54cc

> * Add link to npm?

Ahh good call! I'll update the Usage section with a bit more information from our README in a bit.

> * Update default theme (or add new default-dark) to use on the website?

Lemme loop @joshgoebel into this question. I think we have enough themes already so I don't think we want to add more. Are you talking about using another theme instead of the one I'm using right now?

> * "News" -> "Releases"?

I'm going to leave this as news, so that we can use that area of the site to post announcements (e.g. the news article about a new website)

> * I like the design of the news, but it doesn't really match the rest of the page

Now the Downloads page and the News page share the same blurred dark background! I hope this makes it have a more consistent feel

> * Why is "Code Samples" such a tiny link in the bottom?

Because I couldn't figure out a way to stuff it in the top nav bar without cluttering it... 😓 I'm open to design ideas!

**Comment by allejo** [comment]
After some quick DNS struggles, our new website is live! 🥳 🎉

Moving forward, please report any new bugs at the [`highlightjs/highlightjs.org` repo](https://github.com/highlightjs/highlightjs.org/issues).

**Comment by isagalaev** [comment]
Congrats, folks!

**Comment by allejo** [comment]
> The download page gives you a ZIP file with absolute paths in it... rendering it unusable on FreeBSD (since they have a really old unzip)...

@mdr1384 thanks for your report! This was reported in #3835 and has been fixed in production 😄

**Comment by wil93** [comment]
FYI the demo page is currently detecting everything as Delphi

**Comment by allejo** [comment]
> FYI the demo page is currently detecting everything as Delphi

Thank you for the report, this is being tracked over at https://github.com/highlightjs/highlightjs.org/issues/7. It's a weird quirk with the way the final website is bundled for production.

**Comment by studyToStudy** [comment]
An error occurs when I choose the language I want.

**Comment by joshgoebel** [comment]
@allejo Can this be closed out now?
